src/read/archive.rs

```diff
@@ -44,10 +46,48 @@ impl<'data, R: ReadRef<'data>> ArchiveFile<'data, R> {
         let magic = data
             .read_bytes(&mut tail, archive::MAGIC.len() as u64)
             .read_error("Invalid archive size")?;
-        if magic != &archive::MAGIC[..] {
+        let mut kind_by_header = ArchiveKind::Unknown;
+        if magic == &archive::AIX_BIG_MAGIC[..] {
+            kind_by_header = ArchiveKind::AixBig;
+        } else if magic != &archive::MAGIC[..] {
             return Err(Error("Unsupported archive identifier"));
         }
 
+        if kind_by_header == ArchiveKind::AixBig {
+            // Parse the Fix Header to get member offset
+            let fixedheader = data
+                .read::<archive::AIXBigFixedHeader>(&mut tail)
+                .read_error("Invalid AIX big archive fixed header")?;
+            let firstchildoff = parse_u64_digits(&fixedheader.firstchildoffset, 10)
+                .read_error("Invalid first child offset")?;
+
+            // Move to firstchild
+            tail = firstchildoff;
+
+            let mut file = ArchiveFile {
+                data,
+                offset: tail,
+                len,
+                kind: kind_by_header,
+                symbols: (0, 0),
+                names: &[],
+            };
+
+            // Both the member table and the global symbol table exist as members of the archive and
+            // are kept at the end of the archive file.
+            let mut gst64off = parse_u64_digits(&fixedheader.globsym64offset, 10)
+                .read_error("Invalid global symbol64 table offset")?;
+            if gst64off == 0 {
```

Comment:
Please reverse the logic of this if statement, so that there is a single return for AIX big archives.